File: hw/arboleda/u3/HW29Strategy/StrategyPattern/Strategy/src/main/java/ec/edu/espe/strategy/controller/BubbleSort.java
package ec.edu.espe.strategy.controller;

/**
 *
 * @author Abner Arboleda, LogicLegion, DCCO-ESPE
 */
public class BubbleSort implements SortingStrategy {

    public BubbleSort() {
    }

    @Override
    public int[] sortData(int[] data) {

        int i, j, temp;
        boolean swapped;
        int size = data.length;
                
        for (i = 0; i < size - 1; i++) {
            swapped = false;
            for (j = 0; j < size - i - 1; j++) {
                if (data[j] > data[j + 1]) {

                    temp = data[j];
                    data[j] = data[j + 1];
                    data[j + 1] = temp;
                    swapped = true;
                }
            }

            if (swapped == false) {
                break;
            }
        }
        return data;
    }

}

File: exams/arboleda/u3/Strategy/controller/BubbleSort.py
from controller import SortStrategy

# Concrete Strategy: Bubble Sort
class BubbleSort(SortStrategy):
    def sort(self, data):
        n = len(data)
        for i in range(n):
            for j in range(0, n - i - 1):
                if data[j] > data[j + 1]:
                    data[j], data[j + 1] = data[j + 1], data[j]
        return data